src/types/__test__/1/editArticle.test.ts

```diff
@@ -0,0 +1,1046 @@
+          node {
+            id
+          }
+        }
+      }
+      tags {
+        id
+        content
+      }
+      sticky
+      state
+      license
+      requestForDonation
+      replyToDonator
+      canComment
+      sensitiveByAuthor
+      sensitiveByAdmin
+      revisionCount
+      versions(input: { first: 1 }) {
+        totalCount
+        edges {
+          node {
+            id
+            ... on ArticleVersion {
+              description
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+describe('edit article', () => {
+  const authorId = '1'
+  const titleOriginal = 'original title'
+  const contentOriginal = 'original content'
+  let articleId: string
+  let articleGlobalId: string
+  beforeEach(async () => {
+    const articleService = new ArticleService(connections)
+    const [{ id: _articleId }] = await articleService.createArticle({
+      title: titleOriginal,
+      content: contentOriginal,
+      authorId,
+    })
+    articleId = _articleId
+    articleGlobalId = toGlobalId({ type: NODE_TYPES.Article, id: articleId })
+  })
+  test('edit article content', async () => {
+    const content = 'my customized content'
+    const server = await testClient({
+      isAuth: true,
+      connections,
+    })
+    const { errors, data } = await server.executeOperation({
+      query: EDIT_ARTICLE,
+      variables: {
+        input: {
+          id: articleGlobalId,
+          content,
+        },
+      },
+    })
+    expect(errors).toBeUndefined()
+    expect(data.editArticle.content).toContain(content)
+    expect(data.editArticle.revisionCount).toBe(1)
+
+    // same content will not update revision count
+    const { errors: errors2, data: data2 } = await server.executeOperation({
+      query: EDIT_ARTICLE,
+      variables: {
+        input: {
+          id: articleGlobalId,
+          content,
+        },
+      },
+    })
+    expect(errors2).toBeUndefined()
+    expect(data2.editArticle.content).toContain(content)
+    expect(data2.editArticle.revisionCount).toBe(1)
+  })
+  test('edit article title', async () => {
+    const title = 'my customized title'
+    const server = await testClient({
+      isAuth: true,
+      connections,
+    })
+    const { errors, data } = await server.executeOperation({
+      query: EDIT_ARTICLE,
+      variables: {
+        input: {
+          id: articleGlobalId,
+          title,
+        },
+      },
+    })
+    expect(errors).toBeUndefined()
+    expect(data.editArticle.title).toBe(title)
+    expect(data.editArticle.revisionCount).toBe(1)
+
+    // same title will not update revision count
+    const { data: sameData } = await server.executeOperation({
+      query: EDIT_ARTICLE,
+      variables: {
+        input: {
+          id: articleGlobalId,
+          title,
+        },
+      },
+    })
+    expect(sameData.editArticle.title).toBe(title)
+    expect(sameData.editArticle.revisionCount).toBe(1)
+
+    // empty string is allowed
```

Comment:
same comment as above cuz [title is required](https://github.com/thematters/matters-server/blob/develop/src/mutations/article/publishArticle.ts#L45-L47), pls confirm